mesheryctl/internal/cli/root/system/context.go

```diff
@@ -0,0 +1,399 @@
+// Copyright 2020 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package system
+
+import (
+	"fmt"
+	"sort"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	configuration     *config.MesheryCtlConfig
+	tempCntxt         = "local"
+	set               = false
+	adapters          = []string{}
+	platform          = ""
+	serverURL         = ""
+	newContext        = ""
+	currContext       string
+	allContext        bool
+	tokenNameLocation = map[string]string{} //maps each token name to its specified location
+)
+
+type contextWithLocation struct {
+	Endpoint      string   `mapstructure:"endpoint,omitempty"`
+	Token         string   `mapstructure:"token,omitempty"`
+	Tokenlocation string   `mapstructure:"token,omitempty" yaml:"token-location,omitempty"`
+	Platform      string   `mapstructure:"platform"`
+	Adapters      []string `mapstructure:"adapters,omitempty"`
+	Channel       string   `mapstructure:"channel,omitempty"`
+	Version       string   `mapstructure:"version,omitempty"`
+}
+
+// createContextCmd represents the create command
+var createContextCmd = &cobra.Command{
+	Use:   "create <context-name>",
+	Short: "Create a new context (a named Meshery deployment)",
+	Long:  `Add a new context to Meshery config.yaml file`,
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		tempCntxt := utils.TemplateContext
+
+		if serverURL != "" {
+			err := utils.ValidateURL(serverURL)
+			if err != nil {
+				return err
+			}
+			tempCntxt.Endpoint = serverURL
+		}
+
+		log.Debug("serverURL: `" + tempCntxt.Endpoint + "`")
+
+		if platform != "" {
+			tempCntxt.Platform = platform
+		}
+
+		if len(adapters) >= 1 {
+			tempCntxt.Adapters = adapters
+		}
+
+		err := config.AddContextToConfig(args[0], tempCntxt, viper.ConfigFileUsed(), set)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("Added `%s` context", args[0])
+		return nil
+	},
+}
+
+// deleteContextCmd represents the delete command
+var deleteContextCmd = &cobra.Command{
+	Use:   "delete <context-name>",
```

Comment:
```suggestion
	Use:   "delete context-name",
```

context-name is an arg required, not optional, <> means optional arguments.

Reply:
Thanks, I think [] is for optional and <> for mandatory . 
Let me know.

Reply:
both [], <> means optional args! I also got to know that very recently.

Reply:
Cool